toolkits/Dockerfile.ml

```diff
@@ -0,0 +1,268 @@
+ARG CPP_DEPS=/tmp/cpp_deps
+ARG PY_DEPS=/tmp/py_deps
+ARG PYTHON_VERSION=3.12
+ARG UBUNTU_VERSION=24.04
+
+ARG TRT_IMAGE_TAG=24.12-py3
+ARG TENSORRT_IMAGE=nvcr.io/nvidia/tensorrt
+
+FROM python:${PYTHON_VERSION}-slim AS python-builder
+
+ARG TORCH_VARIANT=cpu
```

Comment:
Don't we need to define those up top in order to be able to change them?

Reply:
not really, as long as it's defined under the correct stage. But it is a question of visibility for someone that wants to read the Dockerfile, so I'm not against putting everything up top too